org.lflang/src/org/lflang/generator/c/CCmakeGenerator.java

```diff
@@ -227,6 +228,22 @@ CodeBuilder generateCMakeCode(
         }
         cMakeCode.newLine();
         
+        // Add the link libraries (if any)
+        for (var lib : targetConfig.linkLibraries) {
+            // Need a special case for libm on Windows
+            if (lib.trim() == "m") {
```

Comment:
This special casing underlines my concern raised in https://github.com/lf-lang/lingua-franca/issues/1120#. As far as we know, this solution only works for the math library (as we have no tests for other libraries), and yet it already needs additional logic to work on all platforms. What happens if we support more libraries than just math? Should we have special cases for all of them?

Reply:
I think that handling special cases in this way is appropriate and could lead to an overall better user experience.  Use of the math library is very common in C programs, and it really unfortunate that the it has to handled differently on different platforms.  But once we understand those needs, why leave it users to have rediscover these quirks themselves?